internal/datastore/crdb/crdb.go

```diff
@@ -165,26 +171,49 @@ func newCRDBDatastore(url string, options ...Option) (datastore.Datastore, error
 		config.maxRevisionStalenessPercent) * time.Nanosecond
 
 	ds := &crdbDatastore{
-		revisions.NewRemoteClockRevisions(
+		RemoteClockRevisions: revisions.NewRemoteClockRevisions(
 			config.gcWindow,
 			maxRevisionStaleness,
 			config.followerReadDelay,
 			config.revisionQuantization,
 		),
-		revision.DecimalDecoder{},
-		url,
-		readPool,
-		writePool,
-		config.watchBufferLength,
-		keyer,
-		config.splitAtUsersetCount,
-		executeWithMaxRetries(config.maxRetries),
-		config.disableStats,
-		changefeedQuery,
+		DecimalDecoder:       revision.DecimalDecoder{},
+		dburl:                url,
+		readPool:             readPool,
+		writePool:            writePool,
+		watchBufferLength:    config.watchBufferLength,
+		writeOverlapKeyer:    keyer,
+		usersetBatchSize:     config.splitAtUsersetCount,
+		disableStats:         config.disableStats,
+		beginChangefeedQuery: changefeedQuery,
 	}
 
 	ds.RemoteClockRevisions.SetNowFunc(ds.headRevisionInternal)
 
+	// TODO: this (and the GC startup that it's based on for mysql/pg) should
+	// be removed and have the lifetimes tied to server start/stop.
+
+	// Start goroutines for pruning
+	if config.enableConnectionBalancing {
+		log.Ctx(initCtx).Info().Msg("starting cockroach connection balancer")
+		ds.pruneCtx, ds.cancelPrune = context.WithCancel(context.Background())
+		ds.pruneGroup, ds.pruneCtx = errgroup.WithContext(ds.pruneCtx)
+		writePoolBalancer := pool.NewNodeConnectionBalancer(ds.writePool, healthChecker, 5*time.Second)
+		readPoolBalancer := pool.NewNodeConnectionBalancer(ds.readPool, healthChecker, 5*time.Second)
+		ds.pruneGroup.Go(func() error {
+			writePoolBalancer.Prune(ds.pruneCtx)
+			return nil
+		})
+		ds.pruneGroup.Go(func() error {
+			readPoolBalancer.Prune(ds.pruneCtx)
+			return nil
+		})
+		ds.pruneGroup.Go(func() error {
+			healthChecker.Poll(ds.pruneCtx, 5*time.Second)
```

Comment:
oh no, I was just curious, I certainly think we should add it when the need arises, but I was more curious to know if you think scenarios where the default won't work.

Reply:
It's possible that you'll run into a problem with scale testing; since this value is hardcoded as you scale up spicedb nodes you're also scaling up the number of probes. So we might need the flag right away.

We may also need a different strategy at scale than having every spicedb pod run a probe (maybe we leader elect which one is the prober, and the rest query for status and/or gossip it piggy-backed on dispatch api calls)